src/CattyTests/Scene/CBSpriteNodePenExtensionTests.swift

```diff
@@ -0,0 +1,161 @@
+            guard let line = node as? LineShapeNode else {
+                XCTFail("Could not cast SKNode to LineShapeNode")
+                return
+            }
+            allShapeNodes.append(line)
+        }
+        return allShapeNodes
+    }
+
+    func testWhenSpritePositionChangedPenUp() {
+        spriteNode.penConfiguration.penDown = false
+        spriteNode.penConfiguration.previousPosition = initialPosition
+        spriteNode.position = CGPoint(x: 1, y: 1)
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWhenSpritePositionChangedPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.penConfiguration.previousPosition = initialPosition
+        spriteNode.position = initialPosition
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.position = CGPoint(x: 1, y: 1)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 2)
+
+        var allShapeNodes = getAllLineShapeNodes()
+
+        XCTAssertEqual(allShapeNodes.count, 1)
+        XCTAssertEqual(allShapeNodes[0].lineWidth, SpriteKitDefines.defaultPenSize, accuracy: CGFloat(Double.epsilon))
+        XCTAssertEqual(allShapeNodes[0].strokeColor, SpriteKitDefines.defaultPenColor)
+        XCTAssertEqual(allShapeNodes[0].pathStartPoint, initialPosition)
+        XCTAssertEqual(allShapeNodes[0].pathEndPoint, spriteNode.position)
+
+        spriteNode.position = CGPoint(x: 2, y: 2)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 3)
+
+        allShapeNodes = getAllLineShapeNodes()
+
+        XCTAssertEqual(allShapeNodes.count, 2)
+        XCTAssertEqual(allShapeNodes[1].pathStartPoint, CGPoint(x: 1, y: 1))
+        XCTAssertEqual(allShapeNodes[1].pathEndPoint, spriteNode.position)
+
+    }
+
+    func testWhenSpritePositionNotChangedPenUp() {
+        spriteNode.penConfiguration.penDown = false
+        spriteNode.penConfiguration.previousPosition = initialPosition
+        spriteNode.position = initialPosition
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWhenSpritePositionNotChangedPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.penConfiguration.previousPosition = initialPosition
+        spriteNode.position = initialPosition
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWithStartMethodPositionChangedAndPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.position = initialPosition
+
+        spriteNode.start(1)
```

Comment:
In this test it is enough to verify whether the `previousPosition` of the `penConfiguration` is set, I think there is no need to verify the actual children since this is already thoroughly tested in the other tests.